liquid-rust-syntax/src/lib.rs

```diff
@@ -41,6 +48,32 @@ pub fn parse_refined_by(tokens: TokenStream, span: Span) -> ParseResult<Vec<Refi
     parse!(RefinedByParser, tokens, span)
 }
 
+pub fn parse_fn_surface_sig(tokens: TokenStream, span: Span) -> ParseResult<ast::FnSig> {
+    let offset = span.lo();
+    let ctx = span.ctxt();
+    let parent = span.parent();
+    let mk_span = |lo: Location, hi: Location| Span::new(lo.0 + offset, hi.0 + offset, ctx, parent);
+    surface_grammar::FnSigParser::new()
+        .parse(&mk_span, Cursor::new(tokens, span.lo()))
+        .map_err(|err| map_err(err, offset, ctx, parent))
+        .map(|sig| surface::desugar(sig))
```

Comment:
We can refactor the macro above to account for this as well

This function becomes

```suggestion
parse!(surface_grammar::FnSigParser, tokens, span)
```

the macro should be refactored to

```rust
macro_rules! parse {
    ($parser:path, $tokens:expr, $span:expr) => {{
        let offset = $span.lo();
        let ctx = $span.ctxt();
        let parent = $span.parent();
        let mk_span =
            |lo: Location, hi: Location| Span::new(lo.0 + offset, hi.0 + offset, ctx, parent);
        <$parser>::new()
            .parse(&mk_span, Cursor::new($tokens, $span.lo()))
            .map_err(|err| map_err(err, offset, ctx, parent))
    }};
}
```
and finally, you have to prepend `grammar` to the parsers in the old functions.